README.rst

```diff
@@ -941,6 +941,15 @@ Choose from the list of available rules:
 
   The correct case must be used for standard PHP types in phpdoc.
 
+* **phpdoc_types_order**
+
+  Sorts PHPDoc types.
+
+  Configuration options:
+
+  - ``null_last`` (``bool``): whether ``null`` should be the last type; defaults to
```

Comment:
Nice start but not customizable enough IMO because on my projects I would like to sort by category of type instead of alphabetically. Another suggestion:

- `sort_algorithm`:
  - `alphabetically` (ignoring leading `\` if any)
  - `by_category` (from "heaviest" to "lightest", i.e.: classes, `object`, `callable`, `resource`, `array<...>`, `...[]`, `array`, scalars, `mixed`, `null`)
  - `none`
- `sort_order`:
  - `ascending`
  - `descending`
- `null_position` (takes precedence over other options):
  - `always_first`
  - `always_last`
  - `default`

Given this discussion, maybe it's better to implement whatever we agree on in this PR directly?